### PR TITLE
Fix AggregateCompanionAdapter::ExtractFunction::setOffset() and test companion functions of array_agg

### PR DESCRIPTION
**Pull request by kagamiori**
Summary:
Before this fix, AggregateCompanionAdapter::ExtractFunction::setOffset() didn't
reserve a uint32_t slot for the tracked row size when allocating groups. This slot
is accessed during aggregations when the UDAF calls trackRowSize(). Not reserving
this slot can cause incorrect result or even crash. This diff fixes this bug and test
companion fucntions of array_agg that would fail without this fix.

Differential Revision: D48421299

### CODE REVIEW

File: velox/exec/AggregateCompanionAdapter.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ void AggregateCompanionAdapter::MergeExtractFunction::extractValues(
 
 int32_t AggregateCompanionAdapter::ExtractFunction::setOffset() const {
   int32_t rowSizeOffset = bits::nbytes(1);
-  int32_t offset = rowSizeOffset;
+  int32_t offset = rowSizeOffset + sizeof(uint32_t);
```
Comment (xiaoxmeng) [inline velox/exec/AggregateCompanionAdapter.cpp:149]: Can you comment this handling?

File: velox/exec/AggregateCompanionAdapter.cpp [hunk 2]
```diff
@@ -146,7 +146,7 @@ void AggregateCompanionAdapter::MergeExtractFunction::extractValues(
 
 int32_t AggregateCompanionAdapter::ExtractFunction::setOffset() const {
   int32_t rowSizeOffset = bits::nbytes(1);
```
Comment (xiaoxmeng) [inline velox/exec/AggregateCompanionAdapter.cpp:148]:
```
const int32_t rowSizeOffset
```

File: velox/exec/tests/SimpleArrayAggAggregate.cpp [hunk 3]
```diff
@@ -148,7 +148,8 @@ exec::AggregateRegistrationResult registerSimpleArrayAggAggregate(
             argTypes.size(), 1, "{} takes at most one argument", name);
         return std::make_unique<SimpleAggregateAdapter<ArrayAggAggregate>>(
             resultType);
-      });
+      },
+      /*registerCompanionFunctions*/ true);
```
Comment (xiaoxmeng) [inline velox/exec/tests/SimpleArrayAggAggregate.cpp:152]:
```
/*registerCompanionFunctions=*/
```

File: velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp [hunk 4]
```diff
@@ -256,7 +256,8 @@ exec::AggregateRegistrationResult registerArray(const std::string& name) {
             argTypes.size(), 1, "{} takes at most one argument", name);
         return std::make_unique<ArrayAggAggregate>(
             resultType, config.prestoArrayAggIgnoreNulls());
-      });
+      },
+      /*registerCompanionFunctions*/ true);
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp:260]: ditto

**Review by xiaoxmeng — APPROVED** [review]
@kagamiori thanks for the fix % nits.

File: velox/exec/tests/SimpleArrayAggAggregate.cpp (hunk 3 above)
Comment (kagamiori) [reply velox/exec/tests/SimpleArrayAggAggregate.cpp:152]: I'd prefer to not have "=" in the comment to be consistent with other places in our codebase. E.g., https://github.com/facebookincubator/velox/blob/d2ebb4e3ba432882d93065eff94d9dad4101bb8b/velox/dwio/dwrf/reader/ColumnReader.cpp#L364-L369

**Pull request closed without merging**